src/enc_effects.c

```diff
@@ -0,0 +1,339 @@
+    [ENCEFF_LEADER_BYRON]       = {ENCEFF_CUTIN_LEADER_BYRON,      SEQ_BATTLE_GYM_LEADER},
+    [ENCEFF_LEADER_VOLKNER]     = {ENCEFF_CUTIN_LEADER_VOLKNER,    SEQ_BATTLE_GYM_LEADER},
+    [ENCEFF_ELITE_FOUR_AARON]   = {ENCEFF_CUTIN_ELITE_FOUR_AARON,  SEQ_BATTLE_ELITE_FOUR},
+    [ENCEFF_ELITE_FOUR_BERTHA]  = {ENCEFF_CUTIN_ELITE_FOUR_BERTHA, SEQ_BATTLE_ELITE_FOUR},
+    [ENCEFF_ELITE_FOUR_FLINT]   = {ENCEFF_CUTIN_ELITE_FOUR_FLINT,  SEQ_BATTLE_ELITE_FOUR},
+    [ENCEFF_ELITE_FOUR_LUCIAN]  = {ENCEFF_CUTIN_ELITE_FOUR_LUCIAN, SEQ_BATTLE_ELITE_FOUR},
+    [ENCEFF_CHAMPION_CYNTHIA]   = {ENCEFF_CUTIN_CHAMPION_CYNTHIA,  SEQ_BATTLE_CHAMPION},
+    [ENCEFF_RIVAL]              = {ENCEFF_CUTIN_USE_LOCAL,         SEQ_BATTLE_RIVAL},
+    [ENCEFF_SHAYMIN]            = {ENCEFF_CUTIN_MYTHICAL,          SEQ_BATTLE_WILD_POKEMON},
+    [ENCEFF_DIALGA_PALKIA]      = {ENCEFF_CUTIN_LEGENDARY,         SEQ_BATTLE_DIALGA_PALKIA},
+    [ENCEFF_UXIE_AZELF]         = {ENCEFF_CUTIN_LEGENDARY,         SEQ_BATTLE_LAKE_GUARDIAN},
+    [ENCEFF_MESPRIT]            = {ENCEFF_CUTIN_USE_LOCAL,         SEQ_BATTLE_LAKE_GUARDIAN},
+    [ENCEFF_ARCEUS]             = {ENCEFF_CUTIN_LEGENDARY,         SEQ_BATTLE_ARCEUS},
+    [ENCEFF_MINOR_LEGENDARIES]  = {ENCEFF_CUTIN_MYTHICAL,          SEQ_BATTLE_LEGENDARY},
+    [ENCEFF_CRESSELIA]          = {ENCEFF_CUTIN_USE_LOCAL,         SEQ_BATTLE_WILD_POKEMON},
+    [ENCEFF_KANTO_BIRDS]        = {ENCEFF_CUTIN_USE_LOCAL,         SEQ_BATTLE_LEGENDARY},
+    [ENCEFF_GIRATINA]           = {ENCEFF_CUTIN_MYTHICAL,          SEQ_BATTLE_GIRATINA},
+    [ENCEFF_REGI_TRIO]          = {ENCEFF_CUTIN_MYTHICAL,          SEQ_BATTLE_REGI_TRIO},
+    [ENCEFF_GALACTIC_GRUNT]     = {ENCEFF_CUTIN_GALACTIC_GRUNT,    SEQ_BATTLE_GALACTIC_GRUNT},
+    [ENCEFF_GALACTIC_CMDR]      = {ENCEFF_CUTIN_GALACTIC_BOSS,     SEQ_BATTLE_GALACTIC_CMDR},
+    [ENCEFF_GALACTIC_CYRUS]     = {ENCEFF_CUTIN_GALACTIC_BOSS,     SEQ_BATTLE_CYRUS},
+    [ENCEFF_FRONTIER]           = {ENCEFF_CUTIN_FRONTIER,          SEQ_BATTLE_TRAINER},
+    [ENCEFF_LINK_BATTLE]        = {ENCEFF_CUTIN_FRONTIER,          SEQ_BATTLE_TRAINER},
+    [ENCEFF_DOUBLE_BATTLE]      = {ENCEFF_CUTIN_DOUBLE,            SEQ_BATTLE_TRAINER},
+    [ENCEFF_DOUBLE_WILD]        = {ENCEFF_CUTIN_DOUBLE,            SEQ_BATTLE_WILD_POKEMON},
+    [ENCEFF_FRONTIER_BRAIN]     = {ENCEFF_CUTIN_FRONTIER,          SEQ_BATTLE_FRONTIER_BRAIN},
+    [ENCEFF_DOUBLE_LEADER]      = {ENCEFF_CUTIN_DOUBLE,            SEQ_BATTLE_GYM_LEADER},
+    [ENCEFF_NORMAL_TRAINER]     = {ENCEFF_CUTIN_USE_LOCAL,         SEQ_BATTLE_TRAINER},
+    [ENCEFF_NORMAL_WILD]        = {ENCEFF_CUTIN_USE_LOCAL,         SEQ_BATTLE_WILD_POKEMON}
+};
+
+static u32 EncEffects_GetEffectPair(const UnkStruct_ov6_02240D5C *battleParams);
+static u32 EncEffects_CutInEffectForPair(u32 param0, const UnkStruct_ov6_02240D5C * param1);
+static u32 EncEffects_BGMForPair(u32 param0, const UnkStruct_ov6_02240D5C * param1);
```

Comment:
I'm not super happy with these names, and I'm open to suggestions for them.

My one caveat is that I wanted `EncEffects_CutInEffect` and `EncEffects_BGM` to be the two external functions.